azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1419 @@
+        name_parts += f", '{arg}'"
+    # TODO: very hacky
+    if "concat(" in name_parts:
+        name_parts = _rem_first_last(name_parts, "'")
+    return f"[resourceId('{rtype}'{name_parts})]"
+
+
+def get_resource_id_by_param(rtype: str, param: TemplateParams) -> str:
+    return f"[resourceId('{rtype}', parameters('{param.value}'))]"
+
+
+def get_ops_extension_name(extension_names: List[str]) -> Optional[str]:
+    for name in extension_names:
+        part = name.rsplit("/", 1)[-1]
+        if part.startswith("azure-iot-operations-") and not part.endswith("platform"):
+            return part
+
+
+class DeploymentContainer:
+    def __init__(
+        self,
+        name: str,
+        api_version: str = "2022-09-01",
+        parameters: Optional[dict] = None,
+        depends_on: Optional[Union[Iterable[str], str]] = None,
+        resource_group: Optional[str] = None,
+        subscription: Optional[str] = None,
+    ):
+        self.name = name
+        self.rcontainer_map: Dict[str, "ResourceContainer"] = {}
+        self.api_version = api_version
+        self.parameters = parameters
+        self.depends_on = depends_on
+        if isinstance(self.depends_on, str):
+            self.depends_on = {self.depends_on}
+        self.resource_group = resource_group
+        self.subscription = subscription
+
+    def add_resources(
+        self,
+        key: Union[StateResourceKey, str],
+        api_version: str,
+        data_iter: Iterable[dict],
+        depends_on: Optional[List[Union[StateResourceKey, str]]] = None,
+        config: Optional[Dict[str, Any]] = None,
+    ):
+        if isinstance(key, StateResourceKey):
+            key = key.value
+        depends_on = process_depends_on(depends_on)
+
+        to_enumerate = list(data_iter)
+
+        count = 0
+        for resource in to_enumerate:
+            count += 1
+            suffix = "" if count <= 1 else f"_{count}"
+            target_key = f"{key}{suffix}"
+
+            self.rcontainer_map[target_key] = ResourceContainer(
+                api_version=api_version,
+                resource_state=resource,
+                depends_on=depends_on,
+                config=config,
+            )
+
+    def get(self):
+        result = {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": self.api_version,
+            "name": self.name,
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": [r.get() for r in list(self.rcontainer_map.values())],
+                },
+            },
+        }
+        if self.resource_group:
+            result["resourceGroup"] = self.resource_group
+        if self.subscription:
+            # TODO: verify
+            # result["subscription"] = self.subscription
```

Comment:
what kind of verification would be needed here?

Reply:
If multi-scope deployments can include a separate subscription than the one being deployed to in the root deployment.